specification/cognitiveservices/OpenAI.Inference/models/completions/streaming/chat_completions.tsp

```diff
@@ -0,0 +1,126 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+
+import "../azure_chat_extensions.tsp";
+import "../common.tsp";
+import "../functions.tsp";
+import "../chat_completions.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+@doc("A single, role-attributed message within a chat completion interaction.")
+model ChatMessageStreamChunk {
+  @doc("The role associated with this message payload.")
+  @projectedName("json", "role")
+  role: ChatRole;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "we explicitly want a nullable string"
+  @doc("The text associated with this message payload.")
+  @projectedName("json", "content")
+  content?: string; // they key might not be present on a streaming scenario
+  // content is optional in the streaming scenario therefore we define as content?: string
+
+  @doc("""
+    The name of the author of this message. `name` is required if role is `function`, and it should be the name of the
+    function whose response is in the `content`. May contain a-z, A-Z, 0-9, and underscores, with a maximum length of
+    64 characters.
+    """)
+  name?: string;
+
+  @doc("The name and arguments of a function that should be called, as generated by the model.")
+  @added(ServiceApiVersions.v2023_07_01_Preview)
+  @projectedName("json", "function_call")
+  functionCall?: FunctionCall;
+
+  @doc("""
+    Additional context data associated with a chat message when requesting chat completions using compatible Azure
+    OpenAI chat extensions. This includes information like the intermediate data source retrievals used to service a
+    request.
+    This context information is only populated when using Azure OpenAI with chat extensions capabilities configured.
+  """)
+  @added(ServiceApiVersions.v2023_08_01_Preview)
+  @projectedName("csharp", "AzureExtensionsContext")
+  context?: AzureChatExtensionsMessageContext;
+}
+
+@doc("""
+The representation of a single prompt completion as part of an overall chat completions request.
+Generally, `n` choices are generated per provided prompt with a default value of 1.
+Token limits and other settings may limit the number of choices generated.
+""")
+model ChatChoiceStreamChunk {
+  @doc("The chat message for a given chat completions prompt.")
+  @projectedName("json", "message")
+  message?: ChatMessage;
```

Comment:
I don't believe `message` can appear in a streamed choice; just `delta`, `finish_reason`, and `choices` (per [OpenAI's YAML again](https://github.com/openai/openai-openapi/blob/e145786e70bf5fc1bc73c7cd19884f445d52c383/openapi.yaml#L3104)) with Azure adding `content_filter_results`.